rfcs/image-optimization.md

```diff
@@ -0,0 +1,276 @@
+---
+date: 2025-12-16
+---
+
+# Image Optimization
+
+## TL;DR
+
+* **We already partially use Vercel image optimization** via `next/image` in layout components (hero, header, featured images, logos).
+* **The biggest current problem** is that **Markdown-rendered content images still use plain `<img>`**, so they:
+  * have no `srcset` / `sizes`
+  * always download the original image
+  * hurt bandwidth, LCP, and overall performance
+* **The most important first step (v1)** is to:
+  * render **all content images with `next/image`**
+  * consistently define correct `sizes` across the app
+* **v2 (optional, later)**: move image transformations & caching from Vercel to Cloudflare (custom loader + R2), if/when we want more control or lower cost.
+
+**Recommendation:**
+👉 **Do v1 first**. It delivers the majority of performance wins with minimal complexity and builds on what we already have.
+
+>[!note]
+> See "Notes" section with a bit of explanation of how things work.
+
+---
+
+## Situation
+
+Flowershow serves all user assets (including images) from Cloudflare R2.
+However, **images are currently rendered via two different pipelines**:
+
+1. **Layout-level images** (hero, header, featured image, logo, etc.)
+   * Rendered with `next/image`
+   * Already benefit from **Vercel’s built-in image optimization**
```

Comment:
how do they benefit from that if on r2? I'm not getting how this is optimized without knowing image size beforehand.

can you say a bit more.